src/pages/workspace/reimburse/WorkspaceReimburseView.js

```diff
@@ -64,140 +64,137 @@ const defaultProps = {
-    constructor(props) {
-        super(props);
-        this.state = {
-            currentRatePerUnit: this.getCurrentRatePerUnitLabel(),
-        };
-    }
-
-    componentDidMount() {
-        this.fetchData();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.policy.customUnits !== this.props.policy.customUnits || prevProps.preferredLocale !== this.props.preferredLocale) {
-            this.setState({currentRatePerUnit: this.getCurrentRatePerUnitLabel()});
-        }
-
-        const reconnecting = prevProps.network.isOffline && !this.props.network.isOffline;
-        if (!reconnecting) {
-            return;
-        }
-
-        this.fetchData();
-    }
-
-    getCurrentRatePerUnitLabel() {
-        const distanceCustomUnit = _.find(lodashGet(this.props, 'policy.customUnits', {}), (unit) => unit.name === 'Distance');
-        const customUnitRate = _.find(lodashGet(distanceCustomUnit, 'rates', {}), (rate) => rate.name === 'Default Rate');
-        const currentUnit = this.getUnitLabel(lodashGet(distanceCustomUnit, 'attributes.unit', 'mi'));
-        const currentRate = this.getRateLabel(customUnitRate);
-        const perWord = this.props.translate('common.per');
+function WorkspaceReimburseView(props) {
+    const [currentRatePerUnit, setCurrentRatePerUnit] = useState('');
+    const viewAllReceiptsUrl = `expenses?policyIDList=${props.policy.id}&billableReimbursable=reimbursable&submitterEmail=%2B%2B`;
+    const distanceCustomUnit = _.find(lodashGet(props.policy, 'customUnits', {}), (unit) => unit.name === 'Distance');
+    const distanceCustomRate = _.find(lodashGet(props.distanceCustomUnit, 'rates', {}), (rate) => rate.name === 'Default Rate');
+    const {translate, toLocaleDigit} = props;
+
+    const getNumericValue = useCallback(
+        (value) => {
+            const numValue = parseFloat(value.toString().replace(toLocaleDigit('.'), '.'));
+            if (Number.isNaN(numValue)) {
+                return NaN;
+            }
+            return numValue.toFixed(3);
+        },
+        [toLocaleDigit],
+    );
+
+    const getRateDisplayValue = useCallback(
+        (value) => {
+            const numValue = getNumericValue(value);
+            if (Number.isNaN(numValue)) {
+                return '';
+            }
+            return numValue.toString().replace('.', toLocaleDigit('.')).substring(0, value.length);
+        },
+        [getNumericValue, toLocaleDigit],
+    );
+
+const getRateLabel = useCallback(
+        (customUnitRate) => getRateDisplayValue(lodashGet(customUnitRate, 'rate', 0) /
+            CONST.POLICY.CUSTOM_UNIT_RATE_BASE_OFFSET),
+        [getRateDisplayValue],
+    );
+
+    const getUnitLabel = useCallback(
+        (value) => translate(`common.${value}`),
+        [translate],
+    );
+
+    const getCurrentRatePerUnitLabel = useCallback(() => {
+        const customUnitRate = _.find(distanceCustomUnit && distanceCustomUnit.rates, (rate) => rate && rate.name === 'Default Rate');
```

Comment:
Let's change this to use `lodashGet` too.

```suggestion
        const customUnitRate = _.find(lodashGet(distanceCustomUnit, 'rates', '{}'), (rate) => rate && rate.name === 'Default Rate');
```